src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -1060,23 +1066,30 @@ private List<Entitlement> findByStackIds(Consumer consumer, List<String> stackId
         return (List<Entitlement>) criteria.list();
     }
 
-    public CandlepinQuery<Entitlement> findByPoolAttribute(Consumer consumer, String attributeName,
+    public List<Entitlement> findByPoolAttribute(Consumer consumer, String attributeName,
         String value) {
 
-        DetachedCriteria criteria = DetachedCriteria.forClass(Entitlement.class)
-            .createAlias("pool", "ent_pool")
-            .createAlias("ent_pool.attributes", "attrs")
-            .add(Restrictions.eq("attrs.indices", attributeName))
-            .add(Restrictions.eq("attrs.elements", value));
+        String jpql = "SELECT e FROM Entitlement e " +
+            "JOIN Pool p on e.pool = p.id " +
+            "JOIN PoolAttribute pa on p.id = pa.poolId " +
+            "WHERE pa.name = :attributeName " +
+            "AND pa.value = :value ";
+        if (consumer != null) {
+            jpql += "AND e.consumer = :consumer";
```

Comment:
This may be a good place for a JPA criteria query instead of string concatenation.